src/lib/support/tests/TestPersistedCounter.cpp

```diff
@@ -185,22 +185,19 @@ static HelpOptions gHelpOptions(TOOL_NAME, "Usage: " TOOL_NAME " [<options...>]\
 
 static OptionSet * gOptionSets[] = { &gHelpOptions, NULL };
 
-int main(int argc, char * argv[])
+extern "C" int TestPersistedCounter(int argc, char * argv[])
```

Comment:
In the future, this should be handled with:

```
#ifdef __cplusplus
extern "C" {
#endif

// C or C++ Declarations

#ifdef __cplusplus
}
#endif
```

wrappers in header files rather than bespoke function decorations such as this.